### DISCUSSION TITLE
Feature Request: Ability to show/hide widgets based on associated product published state

### DISCUSSION THREAD

**Discussion by bendixon-zelis** [Azure/api-management-developer-portal#2627] [Feature request]
The homepage of our developer portal displays an array of cards advertising our various API products. Currently these cards are created manually in the Developer Portal editor. One problem we face is we cannot control the display of the card based on the published status of its associated product so we cannot release Dev Portal changes with cards until we are ready to publish the product and we cannot unpublish a product that has a card without deploying Dev Portal again.

 We have considered introducing a custom widget display the cards based on the published state, but it would be nice if there was a built-in feature, similar to the access control by group feature, that would display or not display a widget based on the published state of a selected product or products.

**Comment by malincrist** [comment]
hello @bendixon-zelis , thanks for requesting this feature!
In the meantime, there is also the default widget `List of products (tiles)` that may be helpful for your situation (similar to list of products, it will display only the published products).

**Comment by bendixon-zelis** [comment]
@malincrist, thank for the pointer. I was guessing we might not be able to use it because of the way we display our tiles depending on whether the user is logged in or not, but I thought I would give it a shot. Maybe I'm doing something wrong but I can't get it display any products as tiles. I tried adding it to the front page and the products page but it only displays "No products found" whereas the products list on the same page displays the products as expected. Included an image below showing the two widgets on the same page.

![image](https://github.com/Azure/api-management-developer-portal/assets/110046405/c84255d4-9abc-4ae0-8243-d79b91efbd1a)

**Comment by malincrist** [comment]
@bendixon-zelis , that is weird.. do you have any errors in the console? if so, can you please share them here as well?
Also, can you please make sure you used the widget `List of Products (tiles)` not another widget? (for example like `API: Products (tiles)`  - which needs an API in order to display the associated products)

**Comment by bendixon-zelis** [comment]
@malincrist, thanks for the help. I had selected the wrong tiling widget. "List of Products (tiles)" displays the products as expected. Unfortunately, I can't swap our current tiles out for this widget because of the differences in UI and limitations in its configuration.

**Comment by sthirthala** [comment]
Is this a managed portal or self hosted portal issue?

**Comment by sthirthala** [comment]
For managed portal, please log support tickets in Azure help+support center

https://learn.microsoft.com/en-us/azure/azure-portal/supportability/how-to-create-azure-support-request
